site/source/tutorials/working_with_virtual_dom.md

```diff
@@ -0,0 +1,106 @@
+
+### Efficiently manage DOM changes
+
+Almost all interactions with a web application cause a visual update of some kind. Many of these updates require the page's DOM structure to be changed by adding, deleting, or moving elements. When this occurs, the web browser must recalculate the position of every element and then re-render them. When performed too often, this process can degrade the performance and responsiveness of the application. Consider this example that is adding a series of items to a list:
+
+```ts
+  const ul = document.getElementById('theList');
+  for (let i = 0; i < 5; i++) {
+    const li = document.createElement('li');
+    li.value = i;
+    li.innerHTML = i;
+    ul.appendChild(li);
+  }
+```
+
+This code will cause the browser to recalculate the flow of the page's elements five times - once per item added to the list. There are ways to eliminate this, using `DocumentFragments` for example, but they often require additional code to be written that obscures the intent of the function. Since an application that uses a virtual DOM doesn't concern itself with intermediate changes, only the final result is required to be generated:
+
+```ts
+  const values = [0, 1, 2, 3, 4];
+
+  v('ul', values.map(value => v('li', [ value ]));
+```
+
+This code returns the `ul` element with the required children in place. When the virtual DOM renders this update, it will determine how to most efficiently render this element.
+
+The virtual DOM achieves its performance improvements by limiting the number of times the DOM is updated and, when it is updated, ensuring that the update is performed as efficiently as possible.
+
+The projector does not update the DOM every time the virtual DOM structure changes. Instead, it informs the browser that an update is required by calling the [requestAnimationFrame](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame) method on the browser's `window` object. This method accepts a callback function that is invoked when the browser is ready to repaint the page as part of its normal rendering process. By waiting until the browser to needs repaint the screen, the virtual DOM library limits the number of times the page must recalculate element positions to one time per rendering cycle.
+
+When the virtual DOM does update the DOM, it does so as efficiently as possible. To achieve the most efficient update possible, projector stores a copy of the virtual nodes used to generate the current DOM structure and creates a mapping between the virtual nodes and the actual ones. When the page is re-rendered, the new virtual DOM is compared to the previous one. The differences between the two structures are used to create an update plan that causes as few changes to the DOM as possible, updating the rendered page. The new virtual DOM nodes are then mapped to the updated DOM and the projector waits for the next update.
```

Comment:
"When the virtual DOM does update the DOM, it does so as efficiently as possible. " Feels like we've said this too many times and we're overselling it now.